examples/model_compress/auto_pruners_torch.py

```diff
@@ -127,65 +130,89 @@ def test(model, device, criterion, val_loader):
+        if args.load_pretrained_model:
+            model.load_state_dict(torch.load(args.pretrained_model_dir))
+            optimizer = torch.optim.SGD(model.parameters(), lr=1e-4, momentum=0.9, weight_decay=5e-4)
+        else:
+            optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9, weight_decay=5e-4)
+            scheduler = MultiStepLR(
+                optimizer, milestones=[int(args.pretrain_epochs*0.5), int(args.pretrain_epochs*0.75)], gamma=0.1)
+    elif args.model == 'resnet50':
+        model = ResNet50().to(device)
+        if args.load_pretrained_model:
+            model.load_state_dict(torch.load(args.pretrained_model_dir))
+            optimizer = torch.optim.SGD(model.parameters(), lr=1e-4, momentum=0.9, weight_decay=5e-4)
+        else:
+            optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9, weight_decay=5e-4)
+            scheduler = MultiStepLR(
+                optimizer, milestones=[int(args.pretrain_epochs*0.5), int(args.pretrain_epochs*0.75)], gamma=0.1)
+    else:
+        raise ValueError("model not recognized")
+
+    if not args.load_pretrained_model:
+        best_acc = 0
+        best_epoch = 0
         for epoch in range(args.pretrain_epochs):
-            train(args, model, device, train_loader,
-                  criterion, optimizer, epoch)
+            train(args, model, device, train_loader, criterion, optimizer, epoch)
             scheduler.step()
-    elif args.model == 'mobilenet_v2':
-        model = models.mobilenet_v2(pretrained=True).to(device)
-
-    if args.save_model:
-        torch.save(model.state_dict(), os.path.join(
-            args.experiment_data_dir, 'model_trained.pth'))
-        print('Model trained saved to %s', args.experiment_data_dir)
+            acc = test(model, device, criterion, val_loader)
+            if acc > best_acc:
+                best_acc = acc
+                best_epoch = epoch
+                state_dict = model.state_dict()
+        model.load_state_dict(state_dict)
+        print('Best acc:', best_acc)
+        print('Best epoch:', best_epoch)
+
+        if args.save_model:
+            torch.save(state_dict, os.path.join(args.experiment_data_dir, 'model_trained.pth'))
+            print('Model trained saved to %s', args.experiment_data_dir)
 
     return model, optimizer
 
 
 def get_dummy_input(args, device):
     if args.dataset == 'mnist':
-        dummy_input = torch.randn(
-            [args.test_batch_size, 1, 28, 28]).to(device)
+        dummy_input = torch.randn([args.test_batch_size, 1, 28, 28]).to(device)
     elif args.dataset in ['cifar10', 'imagenet']:
-        dummy_input = torch.randn(
-            [args.test_batch_size, 3, 32, 32]).to(device)
-
+        dummy_input = torch.randn([args.test_batch_size, 3, 32, 32]).to(device)
     return dummy_input
 
 
+def get_input_size(dataset):
+    if dataset == 'mnist':
+        input_size = (1, 1, 28, 28)
+    elif dataset in ['cifar10', 'imagenet']:
```

Comment:
why input size for imagenet is 32x32?  do we resize the image for imagenet?

Reply:
There is an error here, changed to 256*256. Imagenet experiment is not performed in this PR.